File: app/src/main/java/com/vmb/financialcalculator/DepositActivity.java
package com.vmb.financialcalculator;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.CheckBox;
import android.widget.EditText;

import static com.vmb.financialcalculator.NumberTextWatcher.GetStringInView;

public class DepositActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_deposit);

        EditText initialSumText = findViewById(R.id.textInitialSum);
        initialSumText.addTextChangedListener(new NumberTextWatcher(initialSumText));
        initialSumText.setText(R.string.deposit_initial_sum);

        EditText interestRateSumText = findViewById(R.id.textInterestRate);
        interestRateSumText.setText(R.string.deposit_interest_rate);

        EditText yearText = findViewById(R.id.textYear);
        yearText.setText(R.string.deposit_year);

        EditText monthText = findViewById(R.id.textMonth);
        monthText.setText(R.string.deposit_month);

        EditText additionalSumText =findViewById(R.id.textAdditionalSum);
        additionalSumText.addTextChangedListener(new NumberTextWatcher(additionalSumText));
        additionalSumText.setText(R.string.deposit_additional_sum);
    }

    public void onClickCalculateDeposit(View view) {

        Deposit deposit = new Deposit();
        deposit.setInitial_sum(GetStringInView(findViewById(R.id.textInitialSum)));
        deposit.setInterest_rate(GetStringInView(findViewById(R.id.textInterestRate)));
        deposit.setYear(GetStringInView(findViewById(R.id.textYear)));
        deposit.setMonth(GetStringInView(findViewById(R.id.textMonth)));
        deposit.setAdditional_sum(GetStringInView(findViewById(R.id.textAdditionalSum)));
        boolean capitalization = ((CheckBox) findViewById(R.id.checkBoxCapitalization)).isChecked();

        deposit.CalculateDeposit(capitalization);

        Intent intent = new Intent(this, DepositResultActivity.class);
        intent.putExtra(Deposit.class.getSimpleName(), deposit);
        startActivity(intent);
    }
}

File: app/src/main/java/com/vmb/financialcalculator/DepositActivity.kt
package com.vmb.financialcalculator

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.CheckBox
import android.widget.EditText
import com.vmb.financialcalculator.NumberTextWatcher.Companion.GetStringInView

class DepositActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_deposit)
        val initialSumText = findViewById<EditText>(R.id.textInitialSum)
        initialSumText.addTextChangedListener(NumberTextWatcher(initialSumText))
        initialSumText.setText(R.string.deposit_initial_sum)
        val interestRateSumText = findViewById<EditText>(R.id.textInterestRate)
        interestRateSumText.setText(R.string.deposit_interest_rate)
        val yearText = findViewById<EditText>(R.id.textYear)
        yearText.setText(R.string.deposit_year)
        val monthText = findViewById<EditText>(R.id.textMonth)
        monthText.setText(R.string.deposit_month)
        val additionalSumText = findViewById<EditText>(R.id.textAdditionalSum)
        additionalSumText.addTextChangedListener(NumberTextWatcher(additionalSumText))
        additionalSumText.setText(R.string.deposit_additional_sum)
    }

    fun onClickCalculateDeposit(view: View?) {
        val deposit = Deposit()
        deposit.setInitial_sum(GetStringInView(findViewById<View>(R.id.textInitialSum)))
        deposit.setInterest_rate(GetStringInView(findViewById<View>(R.id.textInterestRate)))
        deposit.setYear(GetStringInView(findViewById<View>(R.id.textYear)))
        deposit.setMonth(GetStringInView(findViewById<View>(R.id.textMonth)))
        deposit.setAdditional_sum(GetStringInView(findViewById<View>(R.id.textAdditionalSum)))
        val capitalization = (findViewById<View>(R.id.checkBoxCapitalization) as CheckBox).isChecked
        deposit.CalculateDeposit(capitalization)
        val intent = Intent(this, DepositResultActivity::class.java)
        intent.putExtra(Deposit::class.java.simpleName, deposit)
        startActivity(intent)
    }
}